File: 백준/Bronze/10798. 세로읽기/세로읽기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main{
    public static void main(String[]args)throws Exception{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        char[][]charArr=new char[5][15];
        int length_max=0;
        for(int i=0;i<5;i++){
            String str=br.readLine();
            int length=str.length();
            if(length_max<length){
                length_max=length;
            }
            for(int j=0;j<length;j++){
                charArr[i][j]=str.charAt(j);
            }
        }
        
        for(int i=0;i<length_max;i++){
            for(int j=0;j<5;j++){
                if(charArr[j][i]==0)continue;
                System.out.print(charArr[j][i]);
            }
        }
        
        
    }
}

File: 백준/Bronze/10798. 세로읽기/세로읽기.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main(){
    val reader = BufferedReader(InputStreamReader(System.`in`))
    val words= Array(5){ reader.readLine()}
    val result=StringBuilder()
    val maxLength=words.maxOf{it.length}

    for(column in 0 until maxLength){
        for(row in 0 until 5){
            if(column<words[row].length){
            result.append(words[row][column])
            }
        }
    }
    println(result.toString())
}